connectors/embeddings-vector-database/src/main/java/io/camunda/connector/embeddingstore/DefaultEmbeddingStoreFactory.java

```diff
@@ -39,6 +60,97 @@ public EmbeddingStore<TextSegment> initializeVectorStore(
           initializeOpenSearchVectorStore(openSearchVectorStore);
       case AmazonManagedOpenSearchVectorStore amazonManagedOpenSearchVectorStore ->
           initializeAmazonManagedOpenSearchVectorStore(amazonManagedOpenSearchVectorStore);
+      case AzureCosmosDbNoSqlVectorStore azureCosmosDbNoSqlVectorStore ->
+          initializeAzureCosmosDbNoSqlVectorStore(azureCosmosDbNoSqlVectorStore, model);
+    };
+  }
+
+  private EmbeddingStore<TextSegment> initializeAzureCosmosDbNoSqlVectorStore(
+      AzureCosmosDbNoSqlVectorStore azureCosmosDbNoSqlVectorStore, EmbeddingModel model) {
+
+    final var cosmosClientBuilder = new CosmosClientBuilder();
+    cosmosClientBuilder.endpoint(azureCosmosDbNoSqlVectorStore.endpoint());
+    cosmosClientBuilder.consistencyLevel(
+        mapConsistencyLevel(azureCosmosDbNoSqlVectorStore.consistencyLevel()));
+    cosmosClientBuilder.contentResponseOnWriteEnabled(true);
+
+    switch (azureCosmosDbNoSqlVectorStore.authentication()) {
+      case AzureCosmosDbNoSqlVectorStore.AzureAuthentication.AzureApiKeyAuthentication apiKey ->
+          cosmosClientBuilder.key(apiKey.apiKey());
+      case AzureCosmosDbNoSqlVectorStore.AzureAuthentication.AzureClientCredentialsAuthentication
+              auth -> {
+        ClientSecretCredentialBuilder clientSecretCredentialBuilder =
+            new ClientSecretCredentialBuilder()
+                .clientId(auth.clientId())
+                .clientSecret(auth.clientSecret())
+                .tenantId(auth.tenantId());
+        if (StringUtils.isNotBlank(auth.authorityHost())) {
+          clientSecretCredentialBuilder.authorityHost(auth.authorityHost());
+        }
+        cosmosClientBuilder.credential(clientSecretCredentialBuilder.build());
+      }
+    }
+
+    final var embedding = new CosmosVectorEmbedding();
+    embedding.setPath(AZURE_COSMOS_DB_VECTOR_EMBEDDING_PATH);
+    embedding.setDataType(CosmosVectorDataType.FLOAT32);
+    embedding.setEmbeddingDimensions(model.dimension());
+    embedding.setDistanceFunction(
+        mapDistanceFunction(azureCosmosDbNoSqlVectorStore.distanceFunction()));
+    final var embeddingPolicy = new CosmosVectorEmbeddingPolicy();
+    embeddingPolicy.setCosmosVectorEmbeddings(List.of(embedding));
+
+    var vectorIndexSpec = new CosmosVectorIndexSpec();
```

Comment:
🔧 Nit: for consistency: 
```suggestion
    final var vectorIndexSpec = new CosmosVectorIndexSpec();
```

Reply:
I wish there was a `val` keyword in Java as the shorthand for `final var` :slightly_smiling_face: